swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/AppLogger.swift

```diff
@@ -9,446 +9,128 @@ import Foundation
     let dateFormatter = ISO8601DateFormatter()
-    dateFormatter.formatOptions = [.withFullDate, .withFullTime, .withFractionalSeconds]
-    self.dateFormatter = dateFormatter
-
     let jsonEncoder = JSONEncoder()
+    dateFormatter.formatOptions = [.withFullDate, .withFullTime, .withFractionalSeconds]
     jsonEncoder.outputFormatting = [.sortedKeys, .withoutEscapingSlashes]
-    self.jsonEncoder = jsonEncoder
-
-    self.newlineData = "\n".data(using: .utf8)!
-
-    workQueue = DispatchQueue(label: "LogWriter.workQueue", qos: .utility)
-    diskLogSwitchingQueue = DispatchQueue(
-      label: "LogWriter.diskLogSwitchingQueue", qos: .background)
-
-    self.logger.error("LogWriter.init: Temporarily using memory log")
-    logDestination = .memory(MemoryLog())
-
-    performInitialSetup()
-  }
-
-  private func performInitialSetup() {
-    diskLogSwitchingQueue.async {
-      let currentIndex: Int = {
-        let currentIndexFromFile: Int? = {
-          guard let currentIndexString = try? String(contentsOf: self.currentIndexFileURL) else {
-            return nil
-          }
-          return Int(currentIndexString)
-        }()
-        if let currentIndexFromFile = currentIndexFromFile {
-          self.logger.error(
-            "LogWriter.performInitialSetup: Current log index read from file: \(currentIndexFromFile, privacy: .public)"
-          )
-          return currentIndexFromFile
-        } else {
-          self.logger.error(
-            "LogWriter.performInitialSetup: Current log index could not be read from file. Assuming current log index as 0."
-          )
-          try? "0".write(to: self.currentIndexFileURL, atomically: true, encoding: .utf8)
-          return 0
-        }
-      }()
-
-      let diskLog = Self.openDiskLog(
-        folderURL: self.folderURL, logFileNameBase: self.logFileNameBase,
-        logIndex: currentIndex, shouldRemoveExistingFile: false,
-        logger: self.logger
-      )
 
-      if diskLog == nil {
-        self.logger.error(
-          "LogWriter.performInitialSetup: Unable to switch log. Log entries will not be written to disk."
-        )
-      }
-
-      self.workQueue.async {
-        if case .memory(let memoryLog) = self.logDestination {
-          let bytesWritten =
-            Self.writeMemoryLogToDisk(memoryLog: memoryLog, diskLog: diskLog, logger: self.logger)
-          self.fileSizeAddendum = UInt64(bytesWritten)
-          self.logDestination = .disk(diskLog)
-          self.logger.error(
-            "LogWriter.performInitialSetup: Switched to disk log."
-          )
-        }
-      }
-    }
-  }
-
-  private static func openDiskLog(
-    folderURL: URL, logFileNameBase: String,
-    logIndex: Int, shouldRemoveExistingFile: Bool, logger: Logger
-  ) -> DiskLog? {
-
-    let fileManager = FileManager.default
-
-    // Look for an existing log file with this log index
-    var existingLogFiles: [URL] = []
-    if let enumerator = fileManager.enumerator(
```

Comment:
@ReactorScram @conectado I believe this may have the thing triggering the App Store privacy notification.